### PR TITLE
Raise when trying to hash an empty file or dir

### PR DESCRIPTION
**Pull request by AurelienGasser**
**Note for reviewers**: I renamed `get_dir_hash` to `get_archive_hash` (more accurate name)

---

The error messages are sometimes confusing 

```
Hash doesn't match ...
```

Most of the times, this error is due to an unexpectedly empty directory. 

This PR improves logging by raising an "Cannot hash empty file/folder" early, instead of raising a "hash doesn't match"-type of exception.

This PR assumes there's no use case when we want to support empty files/directories. If that assumption is incorrect, we could add a boolean option such as `raise_if_empty` to the `get_dir_hash` function

### CODE REVIEW

**Comment by AurelienGasser** [comment]
end to end tests: :heavy_check_mark:

**Comment by inalgnu** [comment]
unit tests are failing, otherwise LGTM

**Comment by AlexandrePicosson** [comment]
LGTM

**Comment by AurelienGasser** [comment]
@inalgnu Mind reviewing again?

File: backend/requirements.txt [hunk 1]
```diff
@@ -29,3 +29,5 @@ uwsgi==2.0.18
 zxcvbn==4.4.28
 boto3==1.11.9
 git+git://github.com/hyperledger/fabric-sdk-py.git@5f6407810f07c6901b10cd3b518dd82aed7482fb
+watchdog==0.10.2
```
Comment (inalgnu) [inline backend/requirements.txt:32]: this seems to be unrelated to this PR, is this wanted?

**Comment by AurelienGasser** [comment]
@inalgnu Ooops I added a commit by mistake. It's ready for review again.
